### PR TITLE
Kernel/riscv64: Implement everything in PageDirectory.cpp

### PR DESCRIPTION
**Pull request by spholz**
Most of this code is just copy-pasted from aarch64 with minor changes for RISC-V.

The next big step to progress further in the boot process is context switching.

### CODE REVIEW

File: Kernel/Arch/riscv64/PageDirectory.cpp [hunk 1]
```diff
@@ -8,39 +8,84 @@
...
+    for (size_t i = 0; i < kernel_pd_index; i++) {
+        directory->m_directory_pages[i] = TRY(MM.allocate_physical_page());
+    }
+
+    // Share the top 1 GiB of kernel-only mappings (>=kernel_mapping_base)
+    directory->m_directory_pages[kernel_pd_index] = MM.kernel_page_directory().m_directory_pages[kernel_pd_index];
+
+    {
+        InterruptDisabler disabler;
+        auto& table = *(PageDirectoryPointerTable*)MM.quickmap_page(*directory->m_directory_table);
```
Comment (kleinesfilmroellchen) [inline Kernel/Arch/riscv64/PageDirectory.cpp:56]: *stares angrily at x86* #22865

**Review by kleinesfilmroellchen — APPROVED** [review]
x86 makes me angry, but this code is good to go.

File: Kernel/Arch/riscv64/PageDirectory.cpp (hunk 1 above)
Comment (Hendiadyoin1) [inline Kernel/Arch/riscv64/PageDirectory.cpp:56]: Would prefer a c++ cast here (`reinterpret_cast` in this case as both are pointers)

File: Kernel/Arch/riscv64/PageDirectory.cpp [hunk 2]
```diff
@@ -8,39 +8,84 @@
...
+        for (size_t i = 0; i < sizeof(m_directory_pages) / sizeof(m_directory_pages[0]); i++) {
+            if (directory->m_directory_pages[i]) {
+                table.raw[i] = (((FlatPtr)directory->m_directory_pages[i]->paddr().as_ptr()) >> PADDR_PPN_OFFSET) << PTE_PPN_OFFSET;
```
Comment (Hendiadyoin1) [inline Kernel/Arch/riscv64/PageDirectory.cpp:59]:
```suggestion
                table.raw[i] = ((directory->m_directory_pages[i]->paddr().get()) >> PADDR_PPN_OFFSET) << PTE_PPN_OFFSET;
```

File: Kernel/Arch/riscv64/PageDirectory.cpp (hunk 1 above)
Comment (spholz) [reply Kernel/Arch/riscv64/PageDirectory.cpp]: I didn't know if I should change this while copy-pasting it from other arches.
But sure, I'll remove those C-style casts.

**Review by Hendiadyoin1 — APPROVED** [review]

**Pull request merged**